### DISCUSSION TITLE
Is there a way to make multiple create3 robots communicate without mixing up their messages?

### DISCUSSION THREAD

**Discussion by bdawkinsroj** [iRobotEducation/create3_docs#590] [Questions] [answered]
### How are you connecting to your Create 3?

Wi-Fi (ROS 2)

### Computer(s) Model(s) and Operating System(s)

Ubuntu 22.04

### Which version of ROS 2 is installed on your computer?

Humble

### Which firmware version is installed on your robot?

H.2.6

### Which RMW is your robot running?

FastRTPS

### Is the robot connected to a network? If so, what is the network type?

Yes; the work wifi

### Are there multiple Create 3 robots connected to your network?

No

### Is multicast enabled?

None

### What is the Adapter Board's USB/BLE Toggle currently switched to?

Bluetooth (default)

### Describe your question.

We were thinking about getting another robot and my boss wanted to see what we could do with communication between multiple robots. I read that the create3's shouldn't talk to eachother as it would mix up messages. I was wondering if there was a way to have them communicate safely.

**Comment by alsora** [comment, marked as answer]
Hi @bdawkinsroj, there are multiple ways to achieve that.

The easiest one is to use ROS_DOMAIN_IDs, every robot has a different number associated and they will be completely isolated from each others.
You can find more details here: https://iroboteducation.github.io/create3_docs/setup/multi-robot/

These settings can be changed from the webserver here: https://iroboteducation.github.io/create3_docs/webserver/application/
